actionview/lib/action_view/template/renderable.rb

```diff
@@ -5,6 +5,10 @@ class Template
     # = Action View Renderable Template for objects that respond to #render_in
     class Renderable # :nodoc:
       def initialize(renderable)
+        unless renderable.respond_to?(:render_in)
+          raise ArgumentError, "'#{renderable.inspect}' is not a renderable object. It must implement #render_in."
```

Comment:
I think its unclear from the PR description what problem this is trying to solve. 

If the goal is just to raise a friendlier error than `NoMethdError` in `render`, then I think we should rescue that when it happens and raise a friendlier message there instead of adding a (slow) `respond_to?` check to the happy path of `Renderable`.

Reply:
I was about to say that. We don't slow down happy path to help developers to not make self-inflected mistakes. If you know about the `: renderable ` object I'm supposing you know that object needs to respond to `render_in`. I'm ok with raising a better message when they don't, but not on the happy path.

Reply:
I've moved this out of the class and back to `TemplateRenderer`.